src/en/config-azure.md

```diff
@@ -1,123 +1,194 @@
+
+ - An Azure account is required. See http://azure.microsoft.com.
+
+ - An SSL/TLS certificate, either an existing one or a new one, will be needed to
+   communicate with Azure.
+
+ - The Juju stable PPA is needed.
+
+ - The Juju client (the host running the below commands) will need the ability
+   to contact the Azure infrastructure on TCP ports 22 and 17070.
+
+### SSL/TLS certificate
+
+The certificate will be uploaded to Azure to enable the Juju client to
+authenticate (and communicate securely) using the associated private key.
+
+Certificate creation is dependant on platform. Below, the *openssl* command
+on Ubuntu is used to create a
+[self-signed certificate](https://en.wikipedia.org/wiki/Self-signed_certificate).
+
+During the creation process, you will be prompted for some information
+(country, state, location, organisation, and common name). 'country' must be a
+valid 2-character value but the others are arbitrary.
+
+!!! Note: We recommend a meaningful name for 'common name' such as 'juju-azure'
+as this will become the "name" of the certificate.
+
+Create the key and certificate now:
 
 ```bash
-juju generate-config
+openssl req -x509 -nodes -days 3650 -newkey rsa:2048 \
+	-keyout juju-azure.pem -out juju-azure.pem
+openssl x509 -inform pem -in juju-azure.pem -outform der -out juju-azure.cer
+chmod 600 juju-azure.pem
 ```
 
-This will generate a file, `environments.yaml`, which will live in your
-`~/.juju/` directory (and will create the directory if it doesn't already
-exist).
+The certificate is in file `juju-azure.cer` and the private key is in
+`juju-azure.pem`.
+
+!!! Note: The PEM file (`juju-azure.pem`) actually contains both the private key
+and the certificate.
 
-**Note:** The above command will not overwrite your existing environments.yaml
-file, or output to stdout. In order to see the boilerplate environments.yaml on
-stdout you need to append the `--show` option. This is helpful if you have an
-existing environments.yaml and just need to add a section. For example:
+Copy the PEM file to where Juju can find it:
+
+```bash
+cp juju-azure.pem /home/ubuntu/.juju
+```
+
+#### Upload certificate to Azure
+
+Log in to the classic console at https://manage.windowsazure.com and in the
+left pane, scroll down and select 'Settings'. Select the 'Management
+Certificates' tab in the right pane. 
+
+![azure_settings_and_certificates](media/config-azure-stable_settings_and_certificates.png)
+
+At the bottom there is an 'Upload' icon.  Use it to upload the certificate
+(`juju-azure.cer`).
+
+
+## Software installation
+
+Proceed to install the software.
+
+```bash
+sudo apt-add-repository -y ppa:juju/stable
+sudo apt-get update
+sudo apt-get install -y juju-core
+```
+
+
+## Configure and bootstrap
+
+If this is a new Juju install then you do not yet have a
+`~/.juju/environments.yaml` file. Create one with
+
+```bash
+juju generate-config
+```
+
+If it does exist (but it was created with an older version of Juju), first move
```

Comment:
the 'older version of juju' reference is irrelevant here. It is either there or not. 
Anyhow, I guess this is moot as the file will be going away, so not to worry